tests/brevitas/nn/quantizers_fixture.py

```diff
@@ -0,0 +1,104 @@
+from brevitas.quant.scaled_int import Int8WeightPerTensorFloat
+from brevitas.quant.scaled_int import Int16Bias
+
+SEED = 123456
+OUT_CH = 16
+IN_CH = 8
+FEATURES = 5
+KERNEL_SIZE = 3
+
+WEIGHT_QUANTIZER = {
+    'None': None,
+    'quant': Int8WeightPerTensorFloat}
+
+ACT_QUANTIZER = {
+    'None': None,
+    'quant': Int8ActPerTensorFloat}
+
+BIAS_QUANTIZER = {
+    'None': None,
+    'quant_external': Int16Bias,
+    'quant_internal': Int8BiasPerTensorFloatInternalScaling}
+
+INPUT_QUANT = {
+    'None': nn.Identity(),
+    'quant': QuantIdentity(return_quant_tensor=True)
+}
+QUANT_WBIOL_IMPL = [
+    QuantLinear, QuantConv1d, QuantConv2d, QuantConvTranspose1d, QuantConvTranspose2d]
+
+
+@pytest_cases.parametrize('bias_quantizer', list(BIAS_QUANTIZER.items()), ids=[f'bias_{c}' for c, _ in BIAS_QUANTIZER.items()])
+@pytest_cases.parametrize('act_quantizer', ACT_QUANTIZER.items(), ids=[f'act_{c}' for c, _ in ACT_QUANTIZER.items()])
+@pytest_cases.parametrize('weight_quantizer', WEIGHT_QUANTIZER.items(), ids=[f'weight_{c}' for c, _ in WEIGHT_QUANTIZER.items()])
+@pytest_cases.parametrize('return_quant_tensor', [True, False], ids=[f'return_quant_tensor_{f}' for f in [True, False]])
+@pytest_cases.parametrize('module', QUANT_WBIOL_IMPL, ids=[f'{c.__name__}' for c in QUANT_WBIOL_IMPL])
+@pytest_cases.parametrize('input_quantizer', INPUT_QUANT.items(), ids=[f'input_{c}' for c, _ in INPUT_QUANT.items()])
+def case_model(weight_quantizer, bias_quantizer, act_quantizer, return_quant_tensor,module, request, input_quantizer):
+    set_case_id(request.node.callspec.id, case_model)
+    case_id  = get_case_id(case_model)
+    _, weight_quantizer = weight_quantizer
+    _, bias_quantizer = bias_quantizer
+    _, act_quantizer = act_quantizer
+    _, input_quantizer = input_quantizer
+
+    impl = case_id.split('-')[2]
+    if module is QuantLinear:
+        layer_kwargs = {
+            'in_features': IN_CH,
+            'out_features': OUT_CH}
+    else:
+        layer_kwargs = {
+            'in_channels': IN_CH,
+            'out_channels': OUT_CH,
+            'kernel_size': KERNEL_SIZE}
+
+    class Model(nn.Module):
+
+        def __init__(self):
+            super().__init__()
+            self.conv = module(
+                **layer_kwargs,
+                bias=True,
+                weight_quant=weight_quantizer,
+                output_quant=act_quantizer,
+                bias_quant=bias_quantizer,
+                return_quant_tensor=return_quant_tensor)
+            self.conv.weight.data.uniform_(-0.01, 0.01)
+
+        def forward(self, x):
+            return self.conv(x)
+
+    torch.random.manual_seed(SEED)
+    module = Model()
+
+    if impl in ('QuantLinear'):
+        in_size = (1, IN_CH)
+    elif impl in ('QuantConv1d', 'QuantConvTranspose1d'):
+        in_size = (1, IN_CH, FEATURES)
+    else:
+        in_size = (1, IN_CH, FEATURES, FEATURES)
+
+    inp = torch.randn(in_size)
+    quant_inp = input_quantizer(inp)
```

Comment:
this should be part of the model definition, so that it switches train/eval with that

Reply:
I left it out because I want to test independently if input is a QuantTensor (full) or a normal Tensor. If it's part of the model definition, it follows return_quant_tensor